src/gluonts/torch/model/simple_feedforward/estimator.py

```diff
@@ -207,27 +203,17 @@ def create_training_data_loader(
         shuffle_buffer_length: Optional[int] = None,
         **kwargs,
     ) -> Iterable:
-        transformation = self._create_instance_splitter(
-            module, "training"
-        ) + SelectFields(TRAINING_INPUT_NAMES)
-
-        training_instances = transformation.apply(
-            Cyclic(data)
-            if shuffle_buffer_length is None
-            else PseudoShuffled(
-                Cyclic(data), shuffle_buffer_length=shuffle_buffer_length
-            )
+        instances = self._create_instance_splitter(module, "training").apply(
+            data, is_train=True
         )
-
-        return IterableSlice(
-            iter(
-                DataLoader(
-                    IterableDataset(training_instances),
-                    batch_size=self.batch_size,
-                    **kwargs,
-                )
-            ),
-            self.num_batches_per_epoch,
+        return data_loader(
+            instances,
+            cycle=True,
+            batch_size=self.batch_size,
+            shuffle_buffer_length=shuffle_buffer_length,
+            field_names=TRAINING_INPUT_NAMES,
+            output_type=torch.tensor,
+            num_batches_per_epoch=self.num_batches_per_epoch,
```

Comment:
Here, `Cyclic` is applied after instance splitting, while prior to the change it was applied before

Reply:
I think it shouldn't matter, since we are not doing any caching.

Reply:
There might be weird edge cases though: if the dataset has just one series, and the splitter happens to sample zero instances, then `Cyclic` will exhaust, see https://github.com/awslabs/gluonts/blob/c6be56264fefe5d879a1fe47ba57798097327533/src/gluonts/itertools.py#L77-L81

Reply:
Should we remove `cycle` from this then and ensure that the passed dataset is already cycled?